invenio_rdm_records/services/access/service.py

```diff
@@ -993,6 +983,14 @@ def delete_grant_by_subject(
         self, identity, id_, subject_id, subject_type, uow=None
     ):
         """Delete an access grant for a record by subject."""
+        # checks if groups are enabled in the instance
+        if subject_type == RecordAccessService.group_subject_type:
+            self.require_permission(
```

Comment:
Honestly, more I see this, more I would not do it checking permissions.
The `groups enabled` feature is a global config flag, that does not depend on the current `identity`.
A simple `config["...ENABLED"]` check will be computationally much faster.